src/read.c

```diff
@@ -2745,9 +2745,6 @@ static avifResult avifDecoderFlush(avifDecoder * decoder)
             return AVIF_RESULT_NO_CODEC_AVAILABLE;
         }
         tile->codec->diag = &decoder->diag;
-        if (!tile->codec->open(tile->codec, decoder)) {
```

Comment:
Is it possible to also postpone the avifCodecCreateInternal() call to avifDecoderNextImage()?

Reply:
That would be more expensive for almost no gain. All calls to `avifCodecCreateInternal()` are little more than a `malloc()`/`memset()` pair, but they set up the existence of the codec in `read.c`. I think this change will give you the flexibility and lazy-init that you're looking for, unless you wanted to pick a _codec_ based on the image's size (which I don't think is a worthwhile use case).